parser.y

```diff
@@ -3634,12 +3634,20 @@ CreateTableStmt:
 |	"CREATE" OptTemporary "TABLE" IfNotExists TableName LikeTableWithOrWithoutParen
 	{
 		$$ = &ast.CreateTableStmt{
-			Table:       $5.(*ast.TableName),
-			ReferTable:  $6.(*ast.TableName),
-			IfNotExists: $4.(bool),
-			IsTemporary: $2.(bool),
+			Table:            $5.(*ast.TableName),
+			ReferTable:       $6.(*ast.TableName),
+			IfNotExists:      $4.(bool),
+			TemporaryKeyword: $2.(ast.TemporaryKeyword),
 		}
 	}
+|	"CREATE" "GLOBAL" "TEMPORARY" "TABLE" IfNotExists TableName TableElementListOpt "ON" "COMMIT" "DELETE" "ROWS"
```

Comment:
Are you going to support `create global temporary table ... like / as` and `TableOption` in another PR?
I think `TableOption` is always a must.

Reply:
yeah why not just replace `OptTemporary` by a rule accepting both `TEMPORARY` and `GLOBAL TEMPORARY` (and &epsilon;).

Reply:
If we make the `OptTemporary` rule accept `GLOBAL TEMPORARY` and reuse the old rules above, we have to disable a lot of things on the TiDB part, it's risky to introduce bugs if we miss something.

So instead of supporting the grammar fully and disable features, I prefer to make a small grammar first and add then features.
@kennytm @djshow832

Reply:
We need to at least support charset, collate, and storage engine before GA. So we need to write duplicate parser code here in the near future.

Reply:
Actually, most of the table options can be ignored ( some of them are already ignored by normal tables ). I think it's fine to handle the other options ( by someone else ).